diff --git a/setup.py b/setup.py
index 4c9e734f..84eb4b3a 100644
--- a/setup.py
+++ b/setup.py
@@ -349,7 +349,7 @@ def get_extensions():
             extra_compile_args=extra_compile_args
         ))
 
-    ffmpeg_exe = distutils.spawn.find_executable('ffmpeg')
+    ffmpeg_exe = distutils.spawn.find_executable('ffmpeg', path=os.path.join(os.getenv('PREFIX'), 'bin'))
     has_ffmpeg = ffmpeg_exe is not None
     # FIXME: Building torchvision with ffmpeg on MacOS or with Python 3.9
     # FIXME: causes crash. See the following GitHub issues for more details.
