From 6e13837f35dd1db1344b41b224370984e13f85e4 Mon Sep 17 00:00:00 2001
From: Deepali Chourasia <deepch23@in.ibm.com>
Date: Thu, 5 Nov 2020 12:35:34 +0000
Subject: [PATCH] Fixed failure on baremetal

---
 setup.py | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index d667446..badf363 100644
--- a/setup.py
+++ b/setup.py
@@ -181,10 +181,12 @@ def get_extensions():
     define_macros = []
 
     extra_compile_args = {}
+    cuda_include_dir = None
     if (torch.cuda.is_available() and ((CUDA_HOME is not None) or is_rocm_pytorch)) \
             or os.getenv('FORCE_CUDA', '0') == '1':
         extension = CUDAExtension
         sources += source_cuda
+        cuda_include_dir = os.getenv("CUDA_INCLUDE_PATH", None)
         if not is_rocm_pytorch:
             define_macros += [('WITH_CUDA', None)]
             nvcc_flags = os.getenv('NVCC_FLAGS', '')
@@ -223,6 +225,8 @@ def get_extensions():
     sources = [os.path.join(extensions_dir, s) for s in sources]
 
     include_dirs = [extensions_dir]
+    if cuda_include_dir:
+        include_dirs.append(cuda_include_dir)
 
     ext_modules = [
         extension(
@@ -365,7 +369,7 @@ def get_extensions():
                     videoapi_src_dir,
                     ffmpeg_include_dir,
                     extensions_dir,
-                ],
+                ] + [cuda_include_dir] if not None else [],
                 library_dirs=[ffmpeg_library_dir] + library_dirs,
                 libraries=[
                     'avcodec',
-- 
1.8.3.1

