diff --git a/setup.py b/setup.py
index fd653fe..28cab6d 100644
--- a/setup.py
+++ b/setup.py
@@ -182,10 +182,12 @@ def get_extensions():
     define_macros = []
 
     extra_compile_args = {}
+    cuda_include_dir = None
     if (torch.cuda.is_available() and ((CUDA_HOME is not None) or is_rocm_pytorch)) \
             or os.getenv('FORCE_CUDA', '0') == '1':
         extension = CUDAExtension
         sources += source_cuda
+        cuda_include_dir = os.getenv("CUDA_INCLUDE_PATH", None)
         if not is_rocm_pytorch:
             define_macros += [('WITH_CUDA', None)]
             nvcc_flags = os.getenv('NVCC_FLAGS', '')
@@ -224,6 +226,8 @@ def get_extensions():
     sources = [os.path.join(extensions_dir, s) for s in sources]
 
     include_dirs = [extensions_dir]
+    if cuda_include_dir:
+        include_dirs.append(cuda_include_dir)
 
     ext_modules = [
         extension(
@@ -408,7 +412,7 @@ def get_extensions():
                     extensions_dir,
                     *ffmpeg_include_dir,
                     *include_dirs
-                ],
+                ] + [cuda_include_dir] if not None else [],
                 library_dirs=ffmpeg_library_dir + library_dirs,
                 libraries=[
                     'avcodec',
