From 04ce6daee812de632f332f5f41533be3173ea418 Mon Sep 17 00:00:00 2001
From: Nishidha Panpaliya <npanpa23@in.ibm.com>
Date: Mon, 2 Nov 2020 03:25:28 -0600
Subject: [PATCH] Fixed baremetal failure

---
 setup.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 85a6921..0f311df 100644
--- a/setup.py
+++ b/setup.py
@@ -122,10 +122,13 @@ def get_extensions():
     define_macros = []
 
     extra_compile_args = {}
+    cuda_include_dir = ""
     if (torch.cuda.is_available() and ((CUDA_HOME is not None) or is_rocm_pytorch)) \
             or os.getenv('FORCE_CUDA', '0') == '1':
         extension = CUDAExtension
         sources += source_cuda
+        cuda_include_dir = os.getenv("CUDA_INCLUDE_PATH", "")
+        print("CUDA Include Path: ", cuda_include_dir)
         if not is_rocm_pytorch:
             define_macros += [('WITH_CUDA', None)]
             nvcc_flags = os.getenv('NVCC_FLAGS', '')
@@ -150,7 +153,8 @@ def get_extensions():
     sources = [os.path.join(extensions_dir, s) for s in sources]
 
     include_dirs = [extensions_dir]
-
+    include_dirs.append(cuda_include_dir)
+    print("Include dirs: ", include_dirs)
     ext_modules = [
         extension(
             'torchvision._C',
@@ -195,6 +199,7 @@ def get_extensions():
                 'torchvision.video_reader',
                 combined_src,
                 include_dirs=[
+                    cuda_include_dir,
                     base_decoder_src_dir,
                     video_reader_src_dir,
                     ffmpeg_include_dir,
-- 
1.8.3.1

