From a9ad5042e17e11bc2912fb7997d7fc736e84c3f6 Mon Sep 17 00:00:00 2001
From: Deepali Chourasia <deepch23@in.ibm.com>
Date: Wed, 10 Mar 2021 11:31:51 +0000
Subject: [PATCH] pick up ancaconda gcc

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index fd653fe..dc1ae78 100644
--- a/setup.py
+++ b/setup.py
@@ -348,7 +348,7 @@ def get_extensions():
         ffmpeg_include_dir = os.path.join(ffmpeg_root, 'include')
         ffmpeg_library_dir = os.path.join(ffmpeg_root, 'lib')
 
-        gcc = distutils.spawn.find_executable('gcc')
+        gcc = distutils.spawn.find_executable(os.getenv("CC"))
         platform_tag = subprocess.run(
             [gcc, '-print-multiarch'], stdout=subprocess.PIPE)
         platform_tag = platform_tag.stdout.strip().decode('utf-8')
-- 
1.8.3.1

